sources/packages/backend/apps/api/src/route-controllers/cas-supplier/_tests_/cas-supplier.aest.controller.getCASSupplier.e2e-spec.ts

```diff
@@ -0,0 +1,96 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeCASSupplier,
+} from "@sims/test-utils";
+import { CASSupplierSiteStatus, SupplierStatus } from "@sims/sims-db";
+
+describe("CASSupplierAESTController(e2e)-getCASSuppliers", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should get all the CAS suppliers for a student when requested given a student id.", async () => {
+    // Arrange
+    const now = new Date();
+    const fakeCASSupplierInfo1 = {
+      supplierStatus: SupplierStatus.Verified,
+      isValid: true,
+      supplierAddress: { supplierSiteCode: "123", lastUpdated: now },
+      supplierNumber: "0001234",
+      supplierProtected: true,
+    };
+    const fakeCASSupplierInfo2 = {
+      supplierStatus: SupplierStatus.PendingSupplierVerification,
+      isValid: false,
+      supplierAddress: {
+        supplierSiteCode: "001",
+        lastUpdated: now,
+        status: "ACTIVE" as CASSupplierSiteStatus,
+        addressLine1: "3350 Douglas St, Victoria BC",
+        siteProtected: "YES",
+      },
```

Comment:
This can be generated by a factory. Related to https://github.com/bcgov/SIMS/pull/3674/files#r1746286636